src/client/client.ts

```diff
@@ -155,6 +155,16 @@ export abstract class SuperfaceClientBase extends Events {
 
   private hookMetrics(): void {
     process.on('beforeExit', () => this.metricReporter?.flush());
+    process.on('uncaughtExceptionMonitor', () => {
+      if (
+        process.rawListeners('uncaughtException').length === 0 &&
+        process.rawListeners('unhandledRejection').length === 0
```

Comment:
I think checking for `unhandledRejection` handlers here doesn't make sense. The exception could be coming from sync code.